client.go

```diff
@@ -492,10 +492,11 @@ func (c *Client) Do(req *Request, resp *Response) error {
 			c.m = m
 		}
 	}
-	hc := m[string(host)]
+	host_str := b2s(host)
```

Comment:
This doesn't work. The string will be used for a map key which means the map will keep a reference to it. But that's unsafe as the underlying bytes might be modified.
See: https://go.dev/play/p/q4yRaxJuq3c

Reply:
What might be an improvement is only doing the `string(host)` conversion once inside `if hc == nil {`. And then doing `hc := m[b2s(host)]` even though I'm not sure if Go allocates for this string cast to access a map, you could test this.

Reply:
Go has not allocated for `m[string(b)]` since 1.3 or so.